setup.php

```diff
@@ -144,10 +144,25 @@ function plugin_init_genericobject() {
       );
 
       // Display a menu entry ?
-      $PLUGIN_HOOKS['menu_toadd']['genericobject'] = array(
-         'config' => 'PluginGenericobjectType',
-         'assets' => 'PluginGenericobjectObject'
-      );
+      if (Session::haveRight('plugin_genericobject_types', READ)) {
+         $PLUGIN_HOOKS['menu_toadd']['genericobject']['config'] = 'PluginGenericobjectType';
+      }
+
+      $object = false;
+      foreach ($DB->request('glpi_profilerights',
```

Comment:
What about adding `right > 0`, do a count on it in the query and rely on that count? That would prevent to loop if not necessary.